pykusto/query.py

```diff
@@ -159,17 +168,29 @@ def summarize(self, *args: Union[AggregationExpression, AssignmentFromAggregatio
         return SummarizeQuery(self, assignments)
 
     def mv_expand(self, *columns: Union[Column, ColumnToType], bag_expansion: BagExpansion = None, with_item_index: Column = None,
-                  limit: int = None):
+                  limit: int = None) -> 'MvExpandQuery':
         if len(columns) == 0:
             raise ValueError("Please specify one or more columns for mv-expand")
         return MvExpandQuery(self, columns, bag_expansion, with_item_index, limit)
 
-    def custom(self, custom_query: str):
+    def custom(self, custom_query: str) -> 'CustomQuery':
         return CustomQuery(self, custom_query)
 
-    # TODO convert python types to kusto types
-    def evaluate(self, udf: FunctionType, type_spec_str: str):
-        return EvaluatePythonQuery(self, udf, type_spec_str)
+    def evaluate(self, plugin_name, *args: ExpressionType, distribution: Distribution = None) -> 'EvaluateQuery':
+        return EvaluateQuery(self, plugin_name, *args, distribution=distribution)
+
+    def evaluate_udf(self, udf: FunctionType, extend: bool = True, distribution: Distribution = None, **type_specs: TypeName) -> 'EvaluateQuery':
```

Comment:
This breaks our interface. My guess is that no one uses this function yet, but still it is worth creating a default implementation.

Reply:
I think this interface is more correct, and it's still early enough to make breaking changes with not much harm